packages/nx/src/plugins/js/project-graph/build-dependencies/target-project-locator.ts

```diff
@@ -259,7 +259,14 @@ export class TargetProjectLocator {
       this.nodes
     );
 
-    return this.packageEntryPointsToProjectMap[dep]?.name ?? null;
+    return (
+      this.packageEntryPointsToProjectMap[dep]?.name ??
+      // if the package exports do not include ".", look for subpath exports
+      Object.entries(this.packageEntryPointsToProjectMap).find(([entryPoint]) =>
+        dep.startsWith(`${entryPoint}/`)
+      )?.[1]?.name ??
```

Comment:
@leosvelperez 
I don't think this part is necessary, now that you've added the logic to ensure the `.` entrypoint is defined in the map.